consensus/XDPoS/XDPoS.go

```diff
@@ -37,8 +37,8 @@ import (
 )
 
 const (
-	ExtraFieldCheck     = false
-	SkipExtraFieldCheck = true
+	ExtraFieldCheck     = true
+	SkipExtraFieldCheck = false
```

Comment:
what's the purpose of this two fields?
the BlockConsensusVersion function also seems to be overly complicated with the implementation. I left comment on the function itself.

Reply:
It's related to this PR https://github.com/XinFinOrg/XDPoSChain/pull/186

Reply:
If we don't put this, synchronization on the special case mentioned in the PR will fail.